nautobot_circuit_maintenance/models.py

```diff
@@ -265,7 +264,14 @@ def get_absolute_url(self):
 
     def to_csv(self):
         """Return fields for bulk view."""
-        return (self.subject, self.provider, self.sender, self.source, self.raw, self.date, self.parsed)
+        return (self.subject, self.provider, self.sender, self.source, self.raw, self.stamp, self.parsed)
+
+    def clean(self):
+        """Add validation when creating a RawNotification."""
+        super().clean()
+
+        if self.stamp > datetime.utcnow():
+            raise ValidationError(f"Stamp time {self.stamp} is not consistent, it's in the future.")
```

Comment:
The more I think about this, the less I like it. If there's clock drift or something I don't want us to reject otherwise valid notifications. Maybe change this to a `logger.warning()` if we need it at all?

Reply:
Ok, I will lower this to just a log message. Not a big deal at all, because the important reference now is the `created`: `since_reference = last_raw_notification.created.timestamp()`